zivid_camera/test/test_zivid_camera.cpp

```diff
@@ -239,79 +246,204 @@ TEST_F(ZividNodeTest, testCapturePublishesTopics)
-    ASSERT_EQ(image->data[index + 1], expectedRGB.g);
-    ASSERT_EQ(image->data[index + 2], expectedRGB.b);
-  };
+  const auto pointCloud = captureViaSDKDefaultSettings();
+  const auto expectedRGBA = pointCloud.copyData<Zivid::ColorRGBA>();
+  ASSERT_EQ(image->width, expectedRGBA.width());
+  ASSERT_EQ(image->height, expectedRGBA.height());
+  for (std::size_t i = 0; i < expectedRGBA.size(); i++)
+  {
+    const auto expected = expectedRGBA(i);
+    const auto index = i * bytes_per_pixel;
+    ASSERT_EQ(image->data[index], expected.r);
+    ASSERT_EQ(image->data[index + 1], expected.g);
+    ASSERT_EQ(image->data[index + 2], expected.b);
+    ASSERT_EQ(image->data[index + 3], expected.a);
+    ASSERT_EQ(expected.a, 255);
+  }
+}
+
+TEST_F(CaptureOutputTest, testCaptureDepthImage)
+{
+  std::optional<sensor_msgs::Image> image;
+  auto depth_image_sub = subscribe<sensor_msgs::Image>(depth_image_topic_name, [&](const auto& i) { image = *i; });
+
+  enableFirst3DAcquisitionAndCapture();
 
-  for (const auto& expectedRGB : miscObjectsExpectedRGBs)
+  ASSERT_TRUE(image.has_value());
+  ASSERT_EQ(image->width, 1920U);
+  ASSERT_EQ(image->height, 1200U);
+  constexpr uint32_t bytes_per_pixel = 4U;
+  ASSERT_EQ(image->step, bytes_per_pixel * image->width);
+  ASSERT_EQ(image->data.size(), image->step * image->height);
+  ASSERT_EQ(image->encoding, "32FC1");
+  ASSERT_EQ(image->is_bigendian, false);
+
+  const auto pointCloud = captureViaSDKDefaultSettings();
+  const auto expectedZ = pointCloud.copyData<Zivid::PointZ>();
+  ASSERT_EQ(image->width, expectedZ.width());
+  ASSERT_EQ(image->height, expectedZ.height());
+  for (std::size_t i = 0; i < expectedZ.size(); i++)
   {
-    verifyPixelColor(expectedRGB);
+    const auto expected = expectedZ(i);
+    const float z = *reinterpret_cast<float*>(image->data.data() + i * bytes_per_pixel);
+    comparePointCoordinate(z, expected.z);
+  }
+}
+
+TEST_F(CaptureOutputTest, testCaptureSNRImage)
+{
+  std::optional<sensor_msgs::Image> image;
+  auto snr_image_sub = subscribe<sensor_msgs::Image>(snr_image_topic_name, [&](const auto& i) { image = *i; });
+
+  enableFirst3DAcquisitionAndCapture();
+
+  ASSERT_TRUE(image.has_value());
+  ASSERT_EQ(image->width, 1920U);
+  ASSERT_EQ(image->height, 1200U);
+  constexpr uint32_t bytes_per_pixel = 4U;
+  ASSERT_EQ(image->step, bytes_per_pixel * image->width);
+  ASSERT_EQ(image->data.size(), image->step * image->height);
+  ASSERT_EQ(image->encoding, "32FC1");
+  ASSERT_EQ(image->is_bigendian, false);
+
+  const auto pointCloud = captureViaSDKDefaultSettings();
+  const auto expectedSNR = pointCloud.copyData<Zivid::SNR>();
+  ASSERT_EQ(image->width, expectedSNR.width());
+  ASSERT_EQ(image->height, expectedSNR.height());
+  for (std::size_t i = 0; i < expectedSNR.size(); i++)
+  {
+    const auto expected = expectedSNR(i);
+    const float snr = *reinterpret_cast<float*>(image->data.data() + i * bytes_per_pixel);
+    ASSERT_EQ(snr, expected.value);
```

Comment:
Why is there a epsilon for the checking the depth image, when it's not needed for the snr here?

Reply:
For points we have converted the output from SDK (mm) to (m) so when we compare we divide SDK output by 1000 and compare it to ROS output.

Reply:
Ahokey, so it's for the numerical error in that division. I'll resolve this now.